components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2746,63 +2739,45 @@ void RewardsServiceImpl::OnRestorePublishersInternal(
   callback(result);
 }
 
-std::unique_ptr<ledger::PublisherInfoList>
-SaveNormalizedPublisherListOnFileTaskRunner(PublisherInfoDatabase* backend,
-    const ledger::PublisherInfoList& list) {
+bool SaveNormalizedPublisherListOnFileTaskRunner(
+    PublisherInfoDatabase* backend,
+    ledger::PublisherInfoList list) {
   if (!backend) {
-    return nullptr;
-  }
-
-  bool success = backend->InsertOrUpdateActivityInfos(list);
-
-  if (!success) {
-    return nullptr;
-  }
-
-  std::unique_ptr<ledger::PublisherInfoList> new_list(
-      new ledger::PublisherInfoList);
-
-  for (auto& publisher : list) {
-    new_list->push_back(publisher);
+    return false;
   }
 
-  return new_list;
+  return backend->InsertOrUpdateActivityInfos(list);
 }
 
 void RewardsServiceImpl::SaveNormalizedPublisherList(
-      const ledger::PublisherInfoListStruct& list) {
-  if (list.list.size() == 0) {
-    std::unique_ptr<ledger::PublisherInfoList> empty_list(
-        new ledger::PublisherInfoList);
-    OnPublisherListNormalizedSaved(std::move(empty_list));
-    return;
+    ledger::PublisherInfoList list) {
+  ContentSiteList site_list;
+  for (const auto& publisher : list) {
+    site_list.push_back(PublisherInfoToContentSite(*publisher));
   }
+  // TODO(bridiver) - this doesn't belong here
+  std::sort(site_list.begin(), site_list.end());
```

Comment:
To solve this one I think we need to emit event to UI that list was normalized so that UI pulls it again. This way we don't need to sort it here, but UI decides if they want to pull. Down side is that data that we already have will need to be pulled from db again

Reply:
actually I think the list normalization itself is a problem. @tmancey and I were discussing because calculated fields shouldn't be persisted in the db

Reply:
the percentage, etc... should be calculated at query time

Reply:
why would we like to do that on query time? then every time that we load rewards page we would nee to recalculate it even though that nothing changed

Reply:
It is normalized every time publisher info is saved which happens very frequently. We are actually wasting far more cycles writing things that are never read. It can also easily lead to incorrect data or complicated data migrations when formulas change

Reply:
also what we're doing in the synopsis normalizer is trivial for the DB to do and doesn't require pulling all the records into memory (a problem for mobile)